ipa-core/src/protocol/prss/mod.rs

```diff
@@ -75,14 +138,20 @@ pub struct IndexedSharedRandomness {
 }
 
 impl SharedRandomness for IndexedSharedRandomness {
-    fn generate_values<I: Into<u128>>(&self, index: I) -> (u128, u128) {
+    fn generate_arrays<I: Into<PrssIndex>, N: ArrayLength>(
+        &self,
+        index: I,
+    ) -> (GenericArray<u128, N>, GenericArray<u128, N>) {
         let index = index.into();
         #[cfg(debug_assertions)]
         {
-            self.used.insert(index);
+            for i in 0..N::USIZE {
+                self.used.insert(index.offset(i));
+            }
```

Comment:
I might suggest that we only need to track against `PrssIndex` for this.  It will greatly reduce the tracking overhead.

Reply:
It's probably sufficient, but aren't we already planning to disable this tracking in production configurations? In which case, I'd rather be cautious and do the more extensive check, unless we determine it's a huge hit to our dev runtimes.

Vectorization will shift usage from the index bits to the offset bits, but the overall number of entries added to the map will stay the same. i.e. this change does not increase the size requirement for the tracking structure.

Reply:
Just a thought.  The API doesn't let you take an index plus a range of offsets that don't start at zero, so I figured... why not.

You might get similar coverage by only checking 0 and N-1, FWIW.